src/adapters/philips_hue/hub.rs

```diff
@@ -0,0 +1,111 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use serde_json;
+use std::sync::Arc;
+use std::thread;
+use std::time::Duration;
+use super::hub_api::HubApi;
+use super::{ HueAction, PhilipsHueAdapter, structs };
+use traits::Controller;
+use uuid::Uuid;
+
+pub struct Hub<C> {
+    pub adapter: PhilipsHueAdapter<C>,
+    pub id: String,
+    pub ip: String,
+    pub api: Arc<HubApi>,
+}
+
+impl<C: Controller> Hub<C> {
+    pub fn new(adapter: PhilipsHueAdapter<C>, id: &str, ip: &str) -> Self {
+        // Get API token from config store, default to a random UUID.
+        let token = adapter.controller.get_config().get_or_set_default(
+            "philips_hue",
+            &format!("token_{}", id),
+            &Uuid::new_v4().to_simple_string());
+        Hub {
+            adapter: adapter,
+            id: id.to_owned(),
+            ip: ip.to_owned(),
+            api: Arc::new(HubApi::new(id, ip, &token)),
+        }
+    }
+    pub fn start(&self) {
+        info!("Starting Hue Hub Service for {}", self.id);
+        let adapter = self.adapter.clone();
+        let id = self.id.clone();
+        let api = self.api.clone();
+
+        thread::spawn(move || {
```

Comment:
This is a stub that will be extended to take over some monitoring function, for example polling sensor values or checking whether lights are going offline.

Stopping these threads will become important for proper teardown, but none of that logic is in place in the adapter management layers, yet.